### PR TITLE
[config] Add new option to enable/disable safety check from developer config toml

### PR DESCRIPTION
**Pull request by arjkesh**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Description
- Allow enabling/disabling of safety check test
- This will allow partner developers to ensure there are no safety vulnerabilities in new containers prior to merge

### Tests run
- Ensured option enables safety scan in PR context

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/test_utils/__init__.py [hunk 1]
```diff
@@ -251,6 +256,10 @@ def is_rc_test_context():
     return sm_remote_tests_val == AllowedSMRemoteConfigValues.RC.value
 
 
+def is_safety_test_context():
```
Comment (arjkesh) [inline test/test_utils/__init__.py:259]: I added this change

File: test/dlc_tests/sanity/test_safety_check.py [hunk 2]
```diff
@@ -149,21 +185,24 @@ def _get_latest_package_version(package):
 @pytest.mark.canary("Run safety tests regularly on production images")
 @pytest.mark.skipif(not is_dlc_cicd_context(), reason="Skipping test because it is not running in dlc cicd infra")
 @pytest.mark.skipif(
-    not (is_mainline_context() or (is_canary_context() and is_time_for_canary_safety_scan())),
+    not (
+        is_mainline_context() or is_safety_test_context() or (is_canary_context() and is_time_for_canary_safety_scan())
```
Comment (arjkesh) [inline test/dlc_tests/sanity/test_safety_check.py:189]: I added this change

File: test/dlc_tests/sanity/quick_checks/test_dlc_developer_config.py [hunk 3]
```diff
@@ -32,6 +32,7 @@ def test_developer_configuration():
     assert config.parse_dlc_developer_configs("test", "eks_tests") is True
     assert config.parse_dlc_developer_configs("test", "ec2_tests") is True
     assert config.parse_dlc_developer_configs("test", "use_scheduler") is False
+    assert config.parse_dlc_developer_configs("test", "safety_check_test") is False
```
Comment (arjkesh) [inline test/dlc_tests/sanity/quick_checks/test_dlc_developer_config.py:35]: I added this change

File: test/dlc_tests/sanity/quick_checks/test_dlc_developer_config.py [hunk 4]
```diff
@@ -50,3 +51,4 @@ def test_developer_config_wrappers_defaults():
     assert config.is_eks_test_enabled() is True
     assert config.is_ec2_test_enabled() is True
     assert config.is_scheduler_enabled() is False
+    assert config.is_safety_check_test_enabled() is False
```
Comment (arjkesh) [inline test/dlc_tests/sanity/quick_checks/test_dlc_developer_config.py:54]: I added this change

File: src/config.py [hunk 5]
```diff
@@ -70,6 +70,10 @@ def is_scheduler_enabled():
     return parse_dlc_developer_configs("test", "use_scheduler")
 
 
+def is_safety_check_test_enabled():
```
Comment (arjkesh) [inline src/config.py:73]: I added this change

File: dlc_developer_config.toml [hunk 6]
```diff
@@ -16,28 +16,27 @@ benchmark_mode = false
 
 [build]
 # Frameworks for which you want to disable both builds and tests
-skip_frameworks = []
+skip_frameworks = ['tensorflow', 'pytorch', 'mxnet', 'huggingface_pytorch']
```
Comment (arjkesh) [inline dlc_developer_config.toml:19]: will revert this change

File: dlc_developer_config.toml [hunk 7]
```diff
@@ -16,28 +16,27 @@ benchmark_mode = false
...
 sanity_tests = true
-ecs_tests = true
-eks_tests = true
-ec2_tests = true
-
-### Off by default
-efa_tests = false
-sagemaker_local_tests = false
+  safety_check_test = false
+ecs_tests = false
```
Comment (arjkesh) [inline dlc_developer_config.toml:29]: will revert these changes

File: dlc_developer_config.toml [hunk 8]
```diff
@@ -26,6 +26,7 @@ do_build = true
 [test]
 ### On by default
 sanity_tests = true
+  safety_check_test = true
```
Comment (arjkesh) [inline dlc_developer_config.toml:29]: default for this will be false

File: dlc_developer_config.toml [hunk 9]
```diff
@@ -26,6 +26,7 @@ do_build = true
 [test]
 ### On by default
```
Comment (arjkesh) [inline dlc_developer_config.toml:27]: Will update this comment so it's not so confusing

File: dlc_developer_config.toml [hunk 10]
```diff
@@ -24,13 +24,13 @@ datetime_tag = true
 do_build = true
 
 [test]
-### On by default
 sanity_tests = true
+  safety_check_test = false
```
Comment (tejaschumbalkar) [inline dlc_developer_config.toml:28]: nit: extra space

**Review by tejaschumbalkar — APPROVED** [review]

File: dlc_developer_config.toml (hunk 10 above)
Comment (arjkesh) [reply dlc_developer_config.toml:28]: this is intentional, just to show that it's an option within sanity tests
